wallet-api/src/Wallet/Generators/Mockchain/StateMachine.hs

```diff
@@ -209,14 +209,15 @@ postcondition _ _ _                                          = Top
 transition :: State a u -> Command a u -> Response u -> State a u
 transition (Initial gm) Initialise S             = Symbolic gm $ initialMockchain gm
 transition (Initial gm) Initialise (C es)        = Concrete es $ gm ^. initialUserState
-transition (Initial _) _ _                       = error "Internal error: non-initial transition, initial state"
-transition _ Initialise _                        = error "Internal error: initial transition, non-initial state"
 transition (Concrete _ a) (UserAction ua) (C es) = Concrete es (uaConcrete ua a)
 transition (Concrete _ a) _ (C es)               = Concrete es a
 transition (Symbolic gm mc) (Transaction _ tx) S = Symbolic gm $ submitTxn mc tx
 transition (Symbolic gm mc) Block S              = Symbolic gm $ validatePending mc
 transition (Symbolic gm mc) (UserAction ua) S    = foldl' (\st cmd -> transition st cmd S) s' (uaCommands ua)
     where s' = Symbolic (over userActions tail gm) mc
+-- The following transitions can never appear due to the state machine precondition
```

Comment:
I actually think it's better to `error` in this case (if you always return the same state then it will eventually loop anyway, doesn't it?)